### PR TITLE
[AMORO-2099] Dashboard Catalog page supports automatically filling of required properties

### LINKED ISSUE #2099: [Subtask]: Dashboard Catalog page supports automatically filling of required properties
### Search before asking

- [X] I have searched in the [issues](https://github.com/NetEase/amoro/issues?q=is%3Aissue) and found no similar issues.

### Description

After select `Metastore` and `Table Format` in the Catalog page, some required properties can be automatically filling to the section `Properties` with empty value, if it doesn't exist, like
image

For different types of `Metastore`, the required properties are different as follows
| Metastore | Type                  | Required Properties |
| ---------- | ---------------- | -------------------- |
| ams           | Internal Catalog | warehouse     |
| hadoop     | External Catalog | warehouse    |
| hive           | External Catalog |    |
| glue           | External Catalog | warehouse,lock-impl,lock-table  |
| custom      | External Catalog | catalog-impl   |

And the Table Format is similar

| Table Format | Required Properties |
| ----------  | -------------------- |
| paimon      | warehouse     |

 I think this requirement can be implemented on the front-end.
### Parent issue

#2007

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by minteliuwm**
## Why are the changes needed?

Close #2099.

## Brief change log

- Dashboard Catalog page supports automatically filling of required properties

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [x] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (no)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: ams/dashboard/src/views/catalogs/Detail.vue [hunk 1]
```diff
@@ -265,6 +265,13 @@ const authConfigMap = {
   'auth.kerberos.krb5': 'Kerberos Krb5'
 }
 
+const defaultPropertiesMap = {
+  ams: 'warehouse',
```
Comment (wangtaohz) [inline ams/dashboard/src/views/catalogs/Detail.vue:269]: It's better to use a list as value to store these properties for each key (ams,hadoop...), in order to support more required properties for each key in the future.

**Review by wangtaohz — COMMENTED** [review]
Thanks for you contribution! I tested it locally and it works well.

However, retaining all previous properties after switching metastore or table format can be confusing, because the user is not sure about whether they are required properties. 

I have an idea to solve this problem, which is to delete any key that was required before the switch but is no longer required after the switch.

File: ams/dashboard/src/views/catalogs/Detail.vue (hunk 1 above)
Comment (minteliuwm) [reply ams/dashboard/src/views/catalogs/Detail.vue]: @wangtaohz  Thanks for your comment, I've updated the code, please review again.

**Review by wangtaohz — APPROVED** [review]
LGTM

**Comment by wangtaohz** [comment]
I tested it locally, it worked well.

image

image

File: ams/dashboard/src/views/catalogs/Detail.vue [hunk 2]
```diff
@@ -313,6 +313,14 @@ const authConfigMap = {
   'auth.kerberos.krb5': 'Kerberos Krb5'
 }
 
+const defaultPropertiesMap = {
+  ams: ['warehouse'],
+  hadoop: ['warehouse'],
+  custom: ['catalog-impl'],
+  glue: ['warehouse', 'lock-impl', 'lock-table'],
```
Comment (XBaith) [inline ams/dashboard/src/views/catalogs/Detail.vue:320]: It should be `lock.table`, refer to https://iceberg.apache.org/docs/latest/configuration/#lock-catalog-properties

Comment (wangtaohz) [reply ams/dashboard/src/views/catalogs/Detail.vue:320]: Thank you for your feedback, I will fix it ASAP.
